cmd/release-controller/sync.go

```diff
@@ -514,6 +517,90 @@ func (c *Controller) syncReady(release *Release) error {
 	return nil
 }
 
+type GitHubPR struct {
+	Org  string
+	Repo string
+	PR   int
+}
+
+type BugzillaPR struct {
+	bugzillaNum int
+	githubPR    GitHubPR
+}
+
+const (
+	GHRegex       = `\(https://github.com/[[:alnum:]]+/[[:alnum:]|\-]+/pull/[\d]+\)`
+	BZRegex       = `\[Bug [\d]+\]`
+	BZAssignRegex = `Assigning the QA contact for review:[[:space:]]+/assign @[[:alnum:]]+`
+)
+
+func getBugzillaPRs(input string) []BugzillaPR {
+	lineRegex := regexp.MustCompile(fmt.Sprintf(`%s.+%s`, BZRegex, GHRegex))
+	bzRegex := regexp.MustCompile(BZRegex)
+	numberRegex := regexp.MustCompile(`[\d]+`)
+	ghRegex := regexp.MustCompile(GHRegex)
+	matches := lineRegex.FindAllString(input, -1)
+	var newArr []BugzillaPR
+	for _, match := range matches {
+		bz := bzRegex.FindString(match)
+		bzID := numberRegex.FindString(bz)
+		bzInt, err := strconv.Atoi(bzID)
+		if err != nil {
+			glog.V(4).Infof("Failed to convert bugzilla ID %s to integer: %v", bzID, err)
+			continue
+		}
+		ghLink := ghRegex.FindString(match)
+		trimmedGHLink := strings.TrimPrefix(ghLink, "(https://github.com/")
+		trimmedGHLink = strings.TrimSuffix(trimmedGHLink, ")")
+		splitGHLink := strings.Split(trimmedGHLink, "/")
+		if len(splitGHLink) != 4 {
+			glog.V(4).Infof("Could not properly split github URL: %s", trimmedGHLink)
+			continue
+		}
+		ghPRInt, err := strconv.Atoi(splitGHLink[3])
+		if err != nil {
+			glog.V(4).Infof("Failed to convert github PR ID %s to integer: %v", splitGHLink[3], err)
+			continue
+		}
+		bzPR := BugzillaPR{
+			bugzillaNum: bzInt,
+			githubPR: GitHubPR{
+				Org:  splitGHLink[0],
+				Repo: splitGHLink[1],
+				PR:   ghPRInt,
+			},
+		}
+		newArr = append(newArr, bzPR)
+	}
+	return newArr
+}
+
+func prApprovedByQA(comments []github.IssueComment) bool {
+	bzRegex := regexp.MustCompile(BZAssignRegex)
+	var lgtms, qaContacts []string
+	for _, comment := range comments {
+		if strings.Contains(comment.Body, "/lgtm") {
```

Comment:
This won't handle the case where the repo allows an approving review to mean LGTM, either -- we should see if we can factor out this logic in Prow and reuse it here

Reply:
Added